### PR TITLE
(176) Request a placement - Existing application flow

### PR DESCRIPTION
**Pull request by richpjames**
# Context
During the 'request a placement' journey if a user responds to the 'reason for placement' question that the reason is 'An additional placement on an existing application' then we ask about arrival dates, duration and reason and then skip a bunch of the questions from the ROTL flow added in #824 and direct them straight to the 'Update to application' page.

[Trello card](https://trello.com/c/ncPgE6LL/176-add-a-flow-to-create-and-submit-a-placement-request)

# Changes in this PR

## Screenshots of UI changes
Placement Applications -- allows me to complete the form for a ROTL request

### CODE REVIEW

File: server/form-pages/placement-application/request-a-placement/additionalPlacementDetails.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+} & ObjectWithDateParts<'arrivalDate'>
+
+@Page({
+  name: 'additional-placement-details',
+  bodyProperties: ['arrivalDate', 'arrivalDate-day', 'arrivalDate-month', 'arrivalDate-year', 'duration', 'reason'],
+})
+export default class AdditionalPlacementDetails implements TasklistPage {
+  title = 'Placement details'
+
+  question = 'Why are you requesting a placement?'
```
Comment (pezholio) [inline server/form-pages/placement-application/request-a-placement/additionalPlacementDetails.ts:19]: Does this need to be here?

File: server/utils/placementRequests/checkYourAnswersUtils.ts [hunk 2]
```diff
@@ -19,9 +18,10 @@ export const getPageTitle = (placementApplication: PlacementApplication, pageNam
   return new Page(placementApplication.data?.['request-a-placement'][pageName], placementApplication).title
 }
 
-export const pagesForReview = () => {
-  const task = PlacementRequest.sections.map(section => section.tasks).flat()[0]
-  return Object.keys(task.pages).filter(page => page !== 'check-your-answers')
+export const pagesForReview = (placementApplication: PlacementApplication) => {
```
Comment (pezholio) [inline server/utils/placementRequests/checkYourAnswersUtils.ts:21]: Bryan made a really good point the other day - if someone goes in and fills out a question, then edits their response to something that alters the branching, the previous response will still show, so we really should be walking the tree to make sure we only return relevant responses, and filter the other ones out. Not for now though, just to be aware.

**Review by pezholio — APPROVED** [review]
LGTM - Good plan on having integration tests for all the journeys. They're small enough for there to be not that much of a hit on build time, but important enough for us to have to have confidence!

File: server/form-pages/placement-application/request-a-placement/additionalPlacementDetails.ts (hunk 1 above)
Comment (richpjames) [reply server/form-pages/placement-application/request-a-placement/additionalPlacementDetails.ts:19]: Nope

File: server/utils/placementRequests/checkYourAnswersUtils.ts (hunk 2 above)
Comment (richpjames) [reply server/utils/placementRequests/checkYourAnswersUtils.ts:21]: Yeah I saw that, great perception.  
Hopefully it won't bite us in the short term

**Pull request merged**